### DISCUSSION TITLE
Questions related to Milvus on scalability, AWS flexibility and offline support?

### DISCUSSION THREAD

**Discussion by Pratyush2758** [milvus-io/milvus#35424] [Q&A and General discussion]
Hi,
We have a requirement of ingesting 1 million policy documents for a project. So in relation to that I wanted some clarity with the following questions:

- Can Milvus be scaled as an enterprise rag level application which can ingest 1 million of particular data?
- Since we are using the confidential data, will an Open source Milvus support the enterprise data hosted in the AWS environment? And if yes, can you provide me with some links to support the argument?
- Also if needed in the future, will Milvus be able to provide the Offline support or not? If yes, the supportive links of docs would be really helpful.

**Comment by yhmo** [comment]
There are already a lot of companies use milvus as vector database for their rag applications. If the policy documents are long-length text, you need some chunking tools to split them into chunks. And you need embedding models to convert the chunks to embeddings and ingest into milvus.
You can deploy milvus by helm or operator:
https://milvus.io/docs/install_cluster-helm.md
https://milvus.io/docs/install_cluster-milvusoperator.md

You can deploy it in AWS, GCP, Azure, etc:
https://milvus.io/docs/eks.md
https://milvus.io/docs/gcp.md
https://milvus.io/docs/azure.md

Milvus supports minio, external s3, azure as storage service:
https://milvus.io/docs/deploy_s3.md#Configure-S3-with-Docker-Compose
https://milvus.io/docs/object_storage_operator.md

Milvus document is online, so far we have no offline document. The online document is a good start for new users.

**Reply by Pratyush2758** [reply]
Thanks for the above response.
So just wanted to specifically reiterate, by the process of chunking and embeddings creation, can Milvus be scaled as an enterprise rag level application and will it be able to handle 1 million documents or more?

**Reply by yhmo** [reply]
Yes, sure. Milvus is a production-ready, highly performant, scalable vector database that handles billions of embeddings. [Zilliz Cloud](https://zilliz.com/) is for enterprise-level vector database, which is powered by Milvus and allows users applications running anywhere with global and multi-cloud reach, built-in data security and high resilience. Both can be integrated in enterprise-level RAG applications.

Search "Milvus RAG" in Google you will find rich materials:
https://milvus.io/docs/build-rag-with-milvus.md
https://medium.com/@zilliz_learn
https://adasci.org/rag-with-milvus-vector-database-and-langchain/
https://medium.com/intel-tech/optimize-vector-databases-enhance-rag-driven-generative-ai-90c10416cb9c

**Comment by Gopichandkothapalli** [comment]
Hi @yhmo 
Thanks for the information above. adding to it and reiterrating again as we looking for specific infromation related to below points-

1. We are planning to deploy a vector database solution for handling approximately 1 million documents, which will scale to multi-million records post-chunking and embedding. we need to undestand how efficiently Milvus can handle the ingestion of around 1 million documents.

2. For compliance with our data confidentiality policies, we require a solution that can be deployed in our own AWS environment. We noticed that Milvus offers both Standalone and Distributed (enterprise-grade) versions. Could you provide more details on deploying the Milvus Distributed version on AWS? Specifically, we are looking for insights on any particular configurations or considerations for an AWS deployment.

3. Given the confidential nature of our data and our requirement to host the vector database within our AWS environment, we would like to know about the support options available from Milvus. If we encounter challenges or issues with Milvus, what kind of support and guidance can Milvus provide? Are there dedicated support for Milvus distributed enterprise users on AWS?

could you please provide your comments and confirmations for above points. which helps us to decide whether to consider milvus or not.

**Reply by xiaofan-luan** [reply]
1. Milvus is known for it's scalability and cost efficiency. The largest deployment we have is 5B on cloud 50B in opensource user’s envrionment.
2. We have open source mills, and also our managed cloud version, see   https://zilliz.com/. There is actually one product can support you to deploy everything in your VPC https://zilliz.com/bring-your-own-cloud, please take a look on it. 
3. We can do both community support or business support. Please contact me at [james.luan@zilliz.com](mailto:james.luan@zilliz.com). and let’s have discussion offline.

**Reply by Gopichandkothapalli** [reply]
Hi @xiaofan-luan ,

Thanks the response, using Milvus via Zilliz cloud is out of option here. we are sololy look to have milvus distributed one deployed in AWS instnace using EKS which was mentioned in the docuemnt "https://milvus.io/docs/prerequisite-helm.md". Regarding this we have some question related to will milvus provide any support on self hosted versions of its, the above mentioned question as well. if you provide us the clarity on them that would be helpful.

**Reply by xiaofan-luan** [reply]
> Hi @xiaofan-luan ,
> 
> Thanks the response, using Milvus via Zilliz cloud is out of option here. we are sololy look to have milvus distributed one deployed in AWS instnace using EKS which was mentioned in the docuemnt "https://milvus.io/docs/prerequisite-helm.md". Regarding this we have some question related to will milvus provide any support on self hosted versions of its, the above mentioned question as well. if you provide us the clarity on them that would be helpful.

Hi Gopichandkothapalli,
Generally speaking we do paid support for oss users, our business is more on the managed service side because we believe this might be best for our customers. We do like to offer necessary community support for your depolyment and if you hit into any issue. It shouldn't be really hard to deploy with helm or operator if you have experience in K8s and cloud infras.

If it's only compliance and data security reasons please take BYOC into considertation. You get all the data in your VPC and we halp you to do the management
